### PR TITLE
[ADD/IMP]Migrated Hotel Management for V10 and Improved migrated files.

### PR DESCRIPTION
**Pull request by KrishnaPrajapati-SerpentCS**

### CODE REVIEW

**Comment by JayVora-SerpentCS** [comment]
Please fix the travis errors

File: board_frontdesk/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+{
+    "name": "Board for Hotel FrontDesk",
+    "version": "0.01",
```
Comment (JayVora-SerpentCS) [inline board_frontdesk/__manifest__.py]: make the version 10.0.0.1

File: board_frontdesk/views/board_frontdesk_view.xml [hunk 2]
```diff
@@ -154,4 +154,4 @@
           parent="menu_hotel_reporting" />
 
 	</data>
-</openerp>
+</odoo>
```
Comment (JayVora-SerpentCS) [inline board_frontdesk/views/board_frontdesk_view.xml:137]: Is <data> tag needed, please check?

File: hotel/__manifest__.py [hunk 3]
```diff
@@ -42,5 +41,4 @@
     'css': ['static/src/css/room_kanban.css'],
     'auto_install': False,
     'installable': True,
-    'application': True,
 }
```
Comment (JayVora-SerpentCS) [inline hotel/__manifest__.py:47]: For Hotel module, we should keep application true

File: hotel/__manifest__.py [hunk 4]
```diff
@@ -22,13 +22,12 @@
...
-    'author': 'Serpent Consulting Services Pvt. Ltd., OpenERP SA',
-    'images': [],
+    'version': '0.07',
+    'author': 'Serpent Consulting Services Pvt. Ltd., OpenERP SA,\
+    Odoo Community Association (OCA)',
     'category': 'Generic Modules/Hotel Management',
     'website': 'http://www.serpentcs.com',
     'depends': ['sale_stock', 'point_of_sale', 'report'],
-    'license': "",
     'demo': ['views/hotel_data.xml'],
```
Comment (JayVora-SerpentCS) [inline hotel/__manifest__.py:33]: License has to be kept as AGPL..Please never ignore license terms.
Follow https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0

File: hotel/models/hotel.py [hunk 5]
```diff
@@ -19,16 +19,15 @@
...
-from openerp import models, fields, api, _
-from openerp import workflow
+from odoo.exceptions import except_orm, UserError, ValidationError
+from odoo.tools import misc, DEFAULT_SERVER_DATETIME_FORMAT
+from odoo import models, fields, api, _
+from odoo import workflow
 from decimal import Decimal
 import datetime
 import urllib2
 import time
```
Comment (JayVora-SerpentCS) [inline hotel/models/hotel.py]: The import order is wrong.
See https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0

File: hotel/report/hotel_report.py [hunk 6]
```diff
@@ -21,38 +21,44 @@
...
-                                [('checkin_date', '>=', date_start),
-                                 ('checkout_date', '<=', date_end)])
-        res = folio_obj.browse(self.cr, self.uid, tids)
-        return res
-
-    def gettotal(self, total):
-        self.temp = self.temp + float(total)
-        return total
+        total_amount = 0.0
+        data_1 = []
```
Comment (JayVora-SerpentCS) [inline hotel/report/hotel_report.py]: Use better variable names

File: hotel/report/hotel_report.py [hunk 7]
```diff
@@ -21,38 +21,44 @@
...
 
-    def getTotal(self):
-        return self.temp
+    @api.model
+    def render_html(self, docids, data=None):
+        self.model = self.env.context.get('active_model')
+        docs = self.env[self.model].browse(self.env.context.get('active_ids', []))
 
+        date_start = data['form'].get('date_start', fields.Date.today())
+        date_end = data['form'].get('date_end', str(datetime.now() + relativedelta(months=+1, day=1, days=-1))[:10])
```
Comment (JayVora-SerpentCS) [inline hotel/report/hotel_report.py]: Tesst this functionality well. I doubt data['form'] is no more needed as you can access wizard fields directly!

File: hotel/views/hotel_view.xml [hunk 8]
```diff
@@ -884,7 +879,6 @@
                                 <group>
                                     <group colspan="4" col="4">
                                         <field name="type" select="2" string="Service Type" />
-                                        <field name="state" select="2" />
                                         <field name="categ_id" domain="[('isservicetype','=',True)]"
                                             select="1" />
```
Comment (JayVora-SerpentCS) [inline hotel/views/hotel_view.xml]: select=1 is needed?

File: hotel/views/report_hotel_management.xml [hunk 9]
```diff
@@ -17,9 +16,9 @@
                     <div class="row mt32 mb32">
                         <div class="text-center">
                             <strong>From:</strong>
-                            <span t-esc="formatLang(data['form']['date_start'],date=True)"/>
+                            <span t-esc="data['date_start']"/>
```
Comment (JayVora-SerpentCS) [inline hotel/views/report_hotel_management.xml]: You should never remove formatLang.. its a special purpose function. Did you make sure in core v10 modules? is this not used anymore?

File: hotel_housekeeping/__manifest__.py [hunk 10]
```diff
@@ -38,7 +36,6 @@
         "report/hotel_housekeeping_report.xml",
         "views/activity_detail.xml",
         "wizard/hotel_housekeeping_wizard.xml",
-        "views/hotel_housekeeping_workflow.xml",
```
Comment (JayVora-SerpentCS) [inline hotel_housekeeping/__manifest__.py:41]: You removed the workflow but made sure to add the same functionality by object and buttons?

File: hotel_housekeeping/models/hotel_housekeeping.py [hunk 11]
```diff
@@ -20,11 +20,11 @@
...
 
-from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
-from openerp import models, fields, api, _
-from openerp.exceptions import ValidationError
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
 import time
-from openerp import workflow
+from odoo import workflow
```
Comment (JayVora-SerpentCS) [inline hotel_housekeeping/models/hotel_housekeeping.py:27]: import order:
core imports first, then odoo imports.

File: hotel_housekeeping/models/hotel_housekeeping.py [hunk 12]
```diff
@@ -92,8 +92,6 @@ def action_set_to_dirty(self):
         @param self: object pointer
         """
         self.write({'state': 'dirty'})
```
Comment (JayVora-SerpentCS) [inline hotel_housekeeping/models/hotel_housekeeping.py]: we can use self.state = 'dirty'!

File: hotel_housekeeping/report/housekeeping_report.py [hunk 13]
```diff
@@ -21,58 +21,58 @@
...
 
 import time
-from openerp import models
-from openerp.report import report_sxw
+from odoo import api, fields, models
 from datetime import datetime
-from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from dateutil.relativedelta import relativedelta
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from dateutil import parser
```
Comment (JayVora-SerpentCS) [inline hotel_housekeeping/report/housekeeping_report.py:29]: import order.

File: hotel_housekeeping/report/housekeeping_report.py [hunk 14]
```diff
@@ -21,58 +21,58 @@
...
-            'time': time,
-            'get_room_no': self.get_room_no,
-            'get_room_activity_detail': self._get_room_activity_detail,
-        })
+class ActivityReport(models.AbstractModel):
+    _name = 'report.hotel_housekeeping.report_housekeeping'
 
-    def _get_room_activity_detail(self, date_start, date_end, room_data):
+    def get_room_activity_detail(self, date_start, date_end, room_no):
         activity_detail = []
```
Comment (JayVora-SerpentCS) [inline hotel_housekeeping/report/housekeeping_report.py:35]: why converted method from private to public?

File: hotel_housekeeping/report/housekeeping_report.py [hunk 15]
```diff
@@ -21,58 +21,58 @@
...
+                            'login': (activity.housekeeper and
+                                      activity.housekeeper.name or ''),
+                            'clean_start_time': activity.clean_start_time,
+                            'clean_end_time': activity.clean_end_time,
+                            'duration': diff})
+            activity_detail.append(act_val)
+        return activity_detail
 
-class ReportLunchorder(models.AbstractModel):
+    @api.model
```
Comment (JayVora-SerpentCS) [inline hotel_housekeeping/report/housekeeping_report.py:63]: Any reason to remove class?

File: hotel_pos_restaurant/__manifest__.py [hunk 16]
```diff
@@ -18,18 +19,16 @@
...
-# ---------------------------------------------------------------------------
+#############################################################################
 
 {
     "name": "Hotel POS Restaurant Management",
-    "version": "0.03",
+    "version": "0.10",
     "author": "Serpent Consulting Services Pvt. Ltd., OpenERP SA,\
     Odoo Community Association (OCA)",
     "category": "Generic Modules/Hotel Restaurant Management",
```
Comment (JayVora-SerpentCS) [inline hotel_pos_restaurant/__manifest__.py:29]: We will not use OCA till its accepted by OCA. So remove from all manifest

File: hotel_pos_restaurant/models/pos_restaurant.py [hunk 17]
```diff
@@ -34,14 +35,14 @@ class HotelFolio(models.Model):
 
     @api.multi
     def action_invoice_create(self, grouped=False, states=None):
-        folio = super(HotelFolio, self)
         state = ['confirmed', 'done']
-        folio = folio.action_invoice_create(grouped=False, states=state)
+        folio = super(HotelFolio, self)
+        invoice_id = folio.action_invoice_create(grouped=False, states=state)
         for line in self:
```
Comment (JayVora-SerpentCS) [inline hotel_pos_restaurant/models/pos_restaurant.py:41]: I did not get the double call to super!

**Comment by KrishnaPrajapati-SerpentCS** [comment]
Hello Sir,

Okay

Thanks

File: hotel_pos_restaurant/models/pos_restaurant.py [hunk 18]
```diff
@@ -34,14 +35,14 @@ class HotelFolio(models.Model):
...
             for pos_order in line.folio_pos_order_ids:
-                pos_order.write({'invoice_id': folio})
+                pos_order.write({'invoice_id': invoice_id})
                 pos_order.action_invoice_state()
```
Comment (JayVora-SerpentCS) [inline hotel_pos_restaurant/models/pos_restaurant.py:44]: you can use pos_order.invoice_id=invoice_id?

File: hotel_pos_restaurant/report/pos_report.py [hunk 19]
```diff
@@ -18,61 +19,73 @@
...
+#############################################################################
 
 import time
-from openerp import models
-from openerp.report import report_sxw
+from odoo import api, fields, models
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from dateutil import parser
```
Comment (JayVora-SerpentCS) [inline hotel_pos_restaurant/report/pos_report.py:30]: correct import order.

File: hotel_pos_restaurant/report/pos_report.py [hunk 20]
```diff
@@ -18,61 +19,73 @@
...
+        tids = folio_obj.search([('checkin_date', '>=', date_start),
                                  ('checkout_date', '<=', date_end)])
-        res = folio_obj.browse(self.cr, self.uid, tids)
         posorder_ids = []
-        for rec in res:
+        for rec in tids:
             if rec.folio_pos_order_ids:
                 posorder_ids.append(rec.folio_pos_order_ids)
-        return posorder_ids
+            return posorder_ids
```
Comment (JayVora-SerpentCS) [inline hotel_pos_restaurant/report/pos_report.py:53]: return seems in wrong indentation

File: hotel_pos_restaurant/views/report_pos_management.xml [hunk 21]
```diff
@@ -17,74 +26,80 @@
...
+	                                <td width="40%"><strong>Customer Name :</strong><span t-field="j.partner_id.name"/></td>
+	                                <td width="30%">&#160;</td>
+		                            <td width="30%"><strong>Room No :</strong><span t-field="j.room_no"/></td>
+	                            </tr>
+	                        </table>
+	                        <br/>
+	                        <table class="table table-bordered">
+		                        <tr>
+		                            <td><strong>Order Number.</strong></td>
+		                            <td><strong>Order Date</strong></td>
```
Comment (JayVora-SerpentCS) [inline hotel_pos_restaurant/views/report_pos_management.xml]: is DOT needed after Order number?

File: hotel_reservation/Doc/ChangeLog.txt [hunk 22]
```diff
@@ -1,15 +1,36 @@
...
+    * Solved warnings and added scheduler to calculate room availability.
+
+0.03 on 2013-11-18 by Sudhir Arya
+    * Added access rights to some objects.
+
+0.02 on 2013-11-13 by Sudhir Arya
+    * Standardized the module as per v7 standards.
+
+1.01 on Oct 1,2013 : Anil Kesariya
+    * Added Access control for manager user Improved label name.
```
Comment (JayVora-SerpentCS) [inline hotel_reservation/Doc/ChangeLog.txt]: Remove all changelog files.

File: hotel_reservation/__manifest__.py [hunk 23]
```diff
@@ -18,18 +19,16 @@
...
-    "version": "0.03",
+    "version": "0.07",
     "author": "Serpent Consulting Services Pvt. Ltd., OpenERP SA,\
     Odoo Community Association (OCA)",
     "category": "Generic Modules/Hotel Reservation",
-    "images": [],
     "website": "http://www.serpentcs.com",
-    "depends": ["hotel", "stock", 'mail'],
-    "license": "",
+    "depends": ["hotel", "stock","mail"],
```
Comment (JayVora-SerpentCS) [inline hotel_reservation/__manifest__.py]: follow pep-8 please.

File: hotel_reservation/i18n/fr.po [hunk 24]
```diff
@@ -0,0 +1,352 @@
+# Translation of OpenERP Server.
+# This file contains the translation of the following modules:
+# 	* hotel_reservation
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: OpenERP Server 5.0.6\n"
```
Comment (JayVora-SerpentCS) [inline hotel_reservation/i18n/fr.po]: Its not openerp server 5.0.6

File: hotel_reservation/models/hotel_reservation.py [hunk 25]
```diff
@@ -130,6 +145,19 @@ class HotelReservation(models.Model):
...
+    def unlink(self):
+        """
+        Overrides orm unlink method.
+        @param self: The object pointer
+        @return: True/False.
+        """
+        for reserv_rec in self:
+            if reserv_rec.state != 'draft':
+                raise ValidationError(_('You can not delete Reservation in %s\
+                state.') % (reserv_rec.state))
```
Comment (JayVora-SerpentCS) [inline hotel_reservation/models/hotel_reservation.py:157]: You can not --> You cannot

File: hotel_reservation/models/hotel_reservation.py [hunk 26]
```diff
@@ -394,19 +432,27 @@ def _create_folio(self):
...
                     res_obj = room_obj.browse([r.id])
                     res_obj.write({'status': 'occupied', 'isroom': False})
             folio_vals.update({'room_lines': folio_lines})
             folio = hotel_folio_obj.create(folio_vals)
+            if folio:
+                for rm_line in folio.room_lines:
+                    rm_line.product_id_change()
             self._cr.execute('insert into hotel_folio_reservation_rel'
                              '(order_id, invoice_id) values (%s,%s)',
                              (reservation.id, folio.id)
```
Comment (JayVora-SerpentCS) [inline hotel_reservation/models/hotel_reservation.py]: Write comments in the method for every little algorithm/logic used.I am sure you can optimise

**Comment by KrishnaPrajapati-SerpentCS** [comment]
yes

**Comment by KrishnaPrajapati-SerpentCS** [comment]
No

**Comment by KrishnaPrajapati-SerpentCS** [comment]
What should be write in license ??

Answer is AGPL ??

It's not working

File: board_frontdesk/__manifest__.py [hunk 27]
```diff
@@ -22,18 +22,16 @@
 
 {
     "name": "Board for Hotel FrontDesk",
-    "version": "0.01",
-    "author": "Serpent Consulting Services Pvt. Ltd.,\
-    Odoo Community Association (OCA)",
+    "version": "10.0.0.1",
+    "author": "Serpent Consulting Services Pvt. Ltd.",
```
Comment (JayVora-SerpentCS) [inline board_frontdesk/__manifest__.py]: "version": "10.0.1.0.0",

File: .travis.yml [hunk 28]
```diff
@@ -12,10 +12,10 @@ addons:
...
 
 env:
-  - VERSION="9.0" LINT_CHECK="1"
-  - VERSION="9.0" ODOO_REPO="odoo/odoo" LINT_CHECK="0"
-  - VERSION="9.0" ODOO_REPO="OCA/OCB" LINT_CHECK="0"
-  - VERSION="9.0" UNIT_TEST="1" LINT_CHECK="0"
+  - VERSION="10.0.0.0.0" LINT_CHECK="1"
+  - VERSION="10.0.0.0.0" ODOO_REPO="odoo/odoo" LINT_CHECK="0"
+  - VERSION="10.0.0.0.0" ODOO_REPO="OCA/OCB" LINT_CHECK="0"
+  - VERSION="10.0.0.0.0" UNIT_TEST="1" LINT_CHECK="0"
```
Comment (JayVora-SerpentCS) [inline .travis.yml:19]: The travis file version is different than manifest one.
Keep VERSION="10.0" LINT_CHECK="1"
Not VERSION="10.0.0.0.0"

Comment (KrishnaPrajapati-SerpentCS) [reply .travis.yml:19]: ok

File: hotel/models/hotel.py [hunk 29]
```diff
@@ -135,6 +134,8 @@ class HotelRoomAmenities(models.Model):
     rcateg_id = fields.Many2one('hotel.room.amenities.type',
                                 'Amenity Catagory')
 
+    product_manager = fields.Many2one('res.users', string='Product Manager')
+
```
Comment (JayVora-SerpentCS) [inline hotel/models/hotel.py:138]: what is the purpose?

File: hotel/models/hotel.py [hunk 30]
```diff
@@ -578,7 +580,7 @@ def onchange_partner_id(self):
                 self.partner_invoice_id = partner_rec.id
                 self.partner_shipping_id = partner_rec.id
                 self.pricelist_id = partner_rec.property_product_pricelist.id
-                raise UserError('Not Any Order For  %s ' % (partner_rec.name))
+                raise _('Not Any Order For  %s ' % (partner_rec.name))
             else:
```
Comment (JayVora-SerpentCS) [inline hotel/models/hotel.py:584]: Why do we remove UserError?

File: hotel/report/hotel_report.py [hunk 31]
```diff
@@ -21,38 +21,60 @@
 # ---------------------------------------------------------------------------
 
 import time
-from openerp import models
-from openerp.report import report_sxw
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from dateutil import parser
+from odoo import api, fields, models
```
Comment (JayVora-SerpentCS) [inline hotel/report/hotel_report.py:28]: are all the above import libs used?

File: hotel/models/hotel.py (hunk 30 above)
Comment (KrishnaPrajapati-SerpentCS) [reply hotel/models/hotel.py:584]: Use  _() so remove UserError Nothing Else

File: hotel/models/hotel.py (hunk 29 above)
Comment (KrishnaPrajapati-SerpentCS) [reply hotel/models/hotel.py:138]: Sir It is amenity manager that manager of amenities & added field in old version so I add that field

File: hotel/report/hotel_report.py (hunk 31 above)
Comment (KrishnaPrajapati-SerpentCS) [reply hotel/report/hotel_report.py:28]: Yes

**Pull request merged**